--- libosinfo-1.5.0/osinfo/osinfo_list.c~	2019-04-11 16:28:11.000000000 +0000
+++ libosinfo-1.5.0/osinfo/osinfo_list.c	2019-07-13 16:36:23.603821832 +0000
@@ -195,7 +195,10 @@
  */
 OsinfoEntity *osinfo_list_get_nth(OsinfoList *list, gint idx)
 {
-    return g_ptr_array_index(list->priv->array, idx);
+    if (idx < osinfo_list_get_length(list))
+	    return g_ptr_array_index(list->priv->array, idx);
+    else
+	    return NULL;
 }
 
 
